web/src/main/java/com/epam/esm/web/controller/PurchaseController.java

```diff
@@ -47,28 +42,29 @@ public PurchaseController(PurchaseService purchaseService, HateoasWrapper hateoa
     }
 
     /**
-     * Find all users
+     * Find all purchases
      *
      * @return list
      */
+    @PreAuthorize("isAuthenticated()")
```

Comment:
As far as I see in your approach, every user can access the purchases of other users. This is not entirely correct. 
Think, a user can have an access only to the purchases he owns. The only exclusion is an admin who can access the purchases of other users.

Reply:
done